src/core/Syntax.jl

```diff
@@ -406,6 +406,90 @@ function functor(types::Tuple, expr::GATExpr;
   invoke_term(theory_type, types, name, term_args...)
 end
 
+abstract type AbstractTheoryMap end
+
+function fmap(T::AbstractTheoryMap, x::GATExpr) end
```

Comment:
Is fmap the right name for this? If we think of functors as containers in the "polynomial functors are containers" sense in Haskell the applying a functor is a generalization of the function `map(f)= list -> [f(x) for x in list]` but we aren't bound by the naming conventions of Haskell. Maybe `fcall`? In this setting the GATExpr is the container so this should be like `gatmap(f::AbstractTheoryMap, x::GATExpr)`

Reply:
I mean, ideally we wouldn't have to name this at all because we would just be overloading the function call syntax. What we call it until we can drop 1.0 support, I don't care, so I'll just follow what you suggest and call it gatmap.

Also, maybe we should rename Catlab to Gatlab :P

Reply:
I think so too!

Reply:
Haha, James made the same joke a while ago!